tee-worker/litentry/core/data-providers/src/geniidata.rs

```diff
@@ -98,18 +98,29 @@ impl GeniidataClient {
 		loop_with_abort_strategy::<fn(&_) -> bool, String, DataProviderError>(
 			addresses,
 			|address| {
-				let query =
-					vec![("limit", GENIIDATA_QUERY_LIMIT), ("offset", "0"), ("address", address)];
-				let response = self
-					.client
-					.get_with::<String, GeniidataResponse>("".to_string(), query.as_slice())
-					.map_err(|e| {
-						DataProviderError::GeniiDataError(format!(
-							"GeniiData response error: {}",
-							e
-						))
-					})?;
-				all_items.extend(response.data.list);
+				let mut offset: u32 = 0;
+				loop {
+					let offset_str = offset.to_string();
+					let query = vec![
+						("limit", GENIIDATA_QUERY_LIMIT),
+						("offset", offset_str.as_str()),
+						("address", address),
+					];
+					let response = self
```

Comment:
It seems that the last round query will have offset > total data count, so we are sure that the response will have meaningful results, right? (instead of error out)

Reply:
Yes, the dp API simply returns empty array (instead error out):

![image](https://github.com/litentry/litentry-parachain/assets/120463031/c3604efa-83cd-4ec1-a964-b6c647308fe4)